Modules/pyexpat.c

```diff
@@ -1881,9 +1959,19 @@ MODULE_INITFUNC(void)
 
     /* export using capsule */
     capi_object = PyCapsule_New(&capi, PyExpat_CAPSULE_NAME, NULL);
-    if (capi_object)
-        PyModule_AddObject(m, "expat_CAPI", capi_object);
-    return m;
+    if (capi_object) {
+        if (PyModule_AddObject(mod, "expat_CAPI", capi_object) < 0) {
+            goto error;
+        }
+    }
+    return 0;
+
+error:
+    Py_CLEAR(state->error);
+    Py_CLEAR(state->xml_parse_type);
```

Comment:
If exec fails, the module is destroyed: it calls module_dealloc() which calls pyexpat_free() which clears these two members.

Reply:
@vstinner good point, I was just thinking that I don't need to do this in the other modules so I shouldn't here either.